### PR TITLE
Add memory leak guide

### PR DESCRIPTION
**Pull request by gturbat**
This will be referenced from more places in the future

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Overview
+
+Profiling has a few datasets to help solve memory leaks, for example Live Heap profile type, which is [available in Python, Go, Node.js and is in beta for Java, .NET, Rust, C and C++][1].
+
+This can be hard to get started with, so Datadog provides an end-to-end guided workflow, starting with Go:
+
+{{< img src="profiler/guide-memory-leak/service-page-memory-leak-walkthrough.mp4" alt="Walk through the memory leak workflow" video=true >}}
+
+## What to expect?
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:21]:
```suggestion
## What to expect
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+<div class="alert alert-info">The workflow requires zero prior knowledge, and is accessible for first time investigations</div>
+
+Through multiple steps, the workflow will:
+1. Scope to the right data
+2. Recommend missing Datadog integrations and upgrades that ease the investigation
+3. Inform about how memory management works in your runtime
+4. Propose potential root causes through [Profile Comparisons][2]
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:29]:
```suggestion
The workflow guides you through several steps to:
1. Scope to the relevant data.
2. Recommend Datadog integrations and upgrades that assist in the investigation.
3. Explain how memory management works in your runtime.
4. Propose potential root causes through [Profile Comparisons][2].
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Requirements
+
+<div class="alert alert-warning">The workflow is in public beta and prone to change. More languages and infrastructure will be supported over time</div>
+
+For now, the workflow needs:
+* a Go service 
+* Running on Kubernetes
+* With [Continuous Profiler enabled][3]
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:38]:
```suggestion
* A Go service running on Kubernetes.
* [Continuous Profiler enabled][3].
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+## How to get started?
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:40]:
```suggestion
## Get started
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:41]:
```suggestion

To investigate a memory leak using the guided workflow:

```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+1. Go to *APM -> Service Page* on the service you want to investigate
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:42]:
```suggestion
1. Go to *APM > Service Page* on the service you want to investigate.
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+2. Click on the tab *Memory Leaks*:
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:43]:
```suggestion
2. Click the *Memory Leaks* tab:
```

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+{{< img src="profiler/guide-memory-leak/service-page-memory-leak-entrypoint.png" alt="Memory Leak workflow entrypoint in the Service Page" style="width:100%;" >}}
+3. Follow the steps
```
Comment (brett0000FF) [inline content/en/profiler/guide/solve-memory-leaks.md:45]:
```suggestion
3. Follow the guided steps to complete your investigation.
```

**Review by brett0000FF — APPROVED** [review]
This is really cool! Let me know if you have any questions about my suggestions.

Side note: Nothing that we need to address right now, but I was curious if you think using the term "workflow" conflicts with our usage in Workflow Automation? I could potentially see this confusing some people, but I don't think it's a big deal. Just something I wanted to mention.

File: content/en/profiler/guide/solve-memory-leaks.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  tag: "Documentation"
+  text: "Datadog Continuous Profiler"
+- link: "/profiler/compare_profiles/"
+  tag: "Documentation"
+  text: "Comparing Profiles"
+---
+
+## Overview
+
+Profiling has a few datasets to help solve memory leaks, for example Live Heap profile type, which is [available in Python, Go, Node.js and is in beta for Java, .NET, Rust, C and C++][1].
```
[the remark this answers is not in the document]
Comment (gturbat) [reply content/en/profiler/guide/solve-memory-leaks.md:15]: My goal was to say you can solve memory leaks for more than just Go, and this guide is just focused on the Go walkthrough feature. But your wording also works!

**Comment by gturbat** [comment]
> This is really cool! Let me know if you have any questions about my suggestions.
> 
> Side note: Nothing that we need to address right now, but I was curious if you think using the term "workflow" conflicts with our usage in Workflow Automation? I could potentially see this confusing some people, but I don't think it's a big deal. Just something I wanted to mention.

Maybe we could use "walkthrough" instead of "workflow"? I have no strong opinion, and I do not know what clicks the more with english speaker :)

**Pull request merged**